src/block/reader.rs

```diff
@@ -382,6 +382,55 @@ impl<R: ChunksReader> SequentialBlockDecompressor<R> {
     }
 }
 
+// note: this is just pseudocode, it will be split up into a lot of separate pieces to allow for customization
+fn read_all_blocks_fully_parallel<Read, PixelBlock>(
+    mut create_reader_at_byte: impl Sync + FnMut(u64) -> Result<Read>,
```

Comment:
with the current approach, the closure must be immutable, because it needs to be sync

Reply:
I don't think requiring the closure to be `Sync` is going to be an issue.